contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+        _;
+    }
+
+    modifier isValidPath(
+        address[] memory _path,
+        address _inputToken,
+        address _outputToken
+    )
+    {
+        if(_inputToken != _outputToken){
+            require(
+                _path[0] == _inputToken || (_inputToken == addresses.weth && _path[0] == DEXAdapter.ETH_ADDRESS),
+                "ExchangeIssuance: INPUT_TOKEN_NOT_IN_PATH"
+            );
+            require(
+                _path[_path.length-1] == _outputToken ||
+                (_outputToken == addresses.weth && _path[_path.length-1] == DEXAdapter.ETH_ADDRESS),
+                "ExchangeIssuance: OUTPUT_TOKEN_NOT_IN_PATH"
+            );
+        }
+        _;
+    }
+
+
+    /* ============ Constructor ============ */
+
+    /**
+    * Sets various contract addresses
+    *
+    * @param _weth                  Address of wrapped native token
+    * @param _quickRouter           Address of quickswap router
+    * @param _sushiRouter           Address of sushiswap router
+    * @param _uniV3Router           Address of uniswap v3 router
+    * @param _uniV3Quoter           Address of uniswap v3 quoter
+    * @param _setController         SetToken controller used to verify a given token is a set
+    * @param _debtIssuanceModule    DebtIssuanceModule used to issue and redeem tokens
+    * @param _compoundLeverageModule    CompoundLeverageModule to sync before every issuance / redemption
+    * @param _aaveAddressProvider   Address of address provider for aaves addresses
+    * @param _curveAddressProvider  Contract to get current implementation address of curve registry
+    * @param _curveCalculator       Contract to calculate required input to receive given output in curve (for exact output swaps)
+    */
+    constructor(
+        address _weth,
+        address _quickRouter,
+        address _sushiRouter,
+        address _uniV3Router,
+        address _uniV3Quoter,
+        IController _setController,
+        IDebtIssuanceModule _debtIssuanceModule,
+        ICompoundLeverageModule _compoundLeverageModule,
+        address _aaveAddressProvider,
+        address _curveAddressProvider,
+        address _curveCalculator
+    )
+    public
+    FlashLoanReceiverBaseV2(_aaveAddressProvider)
+    {
+        setController = _setController;
+        debtIssuanceModule = _debtIssuanceModule;
+        compoundLeverageModule = _compoundLeverageModule;
+
+        addresses.weth = _weth;
+        addresses.quickRouter = _quickRouter;
+        addresses.sushiRouter = _sushiRouter;
+        addresses.uniV3Router = _uniV3Router;
+        addresses.uniV3Quoter = _uniV3Quoter;
+        addresses.curveAddressProvider = _curveAddressProvider;
+        addresses.curveCalculator = _curveCalculator;
+
+        address _cEtherADDRESS = CompoundLeverageModuleStorage(address(_compoundLeverageModule)).underlyingToCToken(_weth);
+        require(_cEtherADDRESS != address(0x0), "CEtherAddress Error");
+        cEtherADDRESS = _cEtherADDRESS;
```

Comment:
Nit: change to `cEtherAddress` for consistency

Reply:
I think it'd also be fine to just pass `_cEtherAddress` in as constructor argument, which would reduce complexity.

Reply:
> Nit: change to `cEtherAddress` for consistency

Seems resolved by 2bc43dfd123d9c826c22e1931048af595bb3ebc3

Reply:
> I think it'd also be fine to just pass `_cEtherAddress` in as constructor argument, which would reduce complexity.

I agree with you on this change and it has not been implemented.